### DISCUSSION TITLE
Inserting large object in Postgresql using jackc/pgx returns "out of memory (SQLSTATE 54000)"

### DISCUSSION THREAD

**Discussion by swapnildaga17** [jackc/pgx#1283] [Q&A] [answered]
I am using [jackc/pgx](https://github.com/jackc/pgx/blob/master/large_objects.go) library to insert largeobjects into Postgres. It works fine when the large objects are small. However in one case the large object was measuring almost 1.8 GB in size. As a result when performing the write operation, there was "out of memory (SQLSTATE 54000)" error.

Here is the code snippet how I am inserting blobs.

```
import (
    "github.com/jackc/pgx/v4"
    "github.com/jackc/pgx/v4/pgxpool"
)

// Read bytes from the file to be imported as large object
b, err := ioutil.ReadFile(pathToLargeObjectFile)
txWrite, err := dbPool.Begin(ctx)
loWrite := txWrite.LargeObjects()

fmt.Printf("Creating new blob with ID : %d", ID)
id, err := loWrite.Create(ctx, ID)
// open blob with ID
obj, err := loWrite.Open(ctx, id, pgx.LargeObjectModeRead|pgx.LargeObjectModeWrite)
n, err := obj.Write(b)
fmt.printf("Written %d byte to blob %d\n", n, id)
```

I get an error on this line

> n, err := obj.Write(b)

How do I prevent the error and successfully import the large object?

I read this post [Inserting Large Object into Postgresql returns 53200 Out of Memory error](https://stackoverflow.com/questions/14509747/inserting-large-object-into-postgresql-returns-53200-out-of-memory-error) which tries to write the bytes in chunks.

Is similar possible with jackc/pgx?

I did not get any response on the same question on [Inserting large object in Postgresql using jackc/pgx returns "out of memory (SQLSTATE 54000)"](https://stackoverflow.com/questions/73277493/inserting-large-object-in-postgresql-using-jackc-pgx-returns-out-of-memory-sql) so asking it here

**Comment by jackc** [comment]
> I read this post [Inserting Large Object into Postgresql returns 53200 Out of Memory error](https://stackoverflow.com/questions/14509747/inserting-large-object-into-postgresql-returns-53200-out-of-memory-error) which tries to write the bytes in chunks.

> Is similar possible with jackc/pgx?

Presumably you could call `obj.Write` multiple times.

**Comment by swapnildaga17** [comment, marked as answer]
Thanks @jackc . 
The obj maintains the current pointer to the end of the previous write. 
So now, I read the file in chunks and write a chunk at a time with the same obj object. Repeating it till the EOF for the file is reached.
